crates/ide-diagnostics/src/handlers/unused_variables.rs

```diff
@@ -20,32 +21,41 @@ pub(crate) fn unused_variables(
         // FIXME: Our infra can't handle allow from within macro expansions rn
         return None;
     }
-    let diagnostic_range = ctx.sema.diagnostics_display_range(ast);
-    let var_name = d.local.primary_source(ctx.sema.db).syntax().to_string();
+    let file_id = ctx.sema.diagnostics_display_range(ast).file_id;
+    let (var_name, text_range) = d
+        .local
+        .primary_source(ctx.sema.db)
+        .name()
+        .map(|v| (v.value.text().to_owned(), v.value.syntax().text_range()))?;
```

Comment:
The text range here might be in a macro expansion instead of a real file. Should be fine to keep the `diagnostic_range` as before